pkg/gomod/parse.go

```diff
@@ -0,0 +1,53 @@
+package gomod
+
+import (
+	"bufio"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+)
+
+// Parse parses a go.sum file
+func Parse(r io.Reader) ([]types.Library, error) {
+	var libs []types.Library
+	tmp := make(map[string]string)
+
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+		s := strings.Fields(line)
+		// go.sum records and sorts all non-major versions
+		// with the latest version as last entry
+		tmp[s[0]] = parseSemVer(s[1])
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	for k, v := range tmp {
+		libs = append(libs, types.Library{
+			Name:    k,
+			Version: v,
+		})
+	}
+
+	return libs, nil
+}
+
+// parseSemVer parses semver from go.sum
+// ignoring pseudo-version
+func parseSemVer(v string) string {
```

Comment:
I think we should keep pseudo-version. Let's say the latest git tag is `v1.0.0`. If there are some commits after `v1.0.0`, the version would be like `v1.0.1-0.20201117152358-0edc412565dc`. Then, the library will release `v1.0.1`. In that case, `v1.0.1` and `v1.0.1-0.20201117152358-0edc412565dc` are different and should be handled differently.

Reply:
that makes sense, updated to keep pseudo version